sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java

```diff
@@ -444,138 +441,98 @@
     }
 
     /**
-     * Set the {@link ConfigurationSetting} to read-only with a matching {@code key}, and optional {@code label}
-     * combination.
+     * Sets the read-only status for the {@link ConfigurationSetting}. If {@code isReadOnly} is {@code true} the
+     * configuration setting will be read-only, {@code false} will clear the read-only status.and optional
+     * {@code label} combination. If the {@code isReadOnly} is true or null, the setting will be set to read-only.
+     * If false, the setting will not be read-only.
      *
      * <p><strong>Code Samples</strong></p>
      *
      * <p>Set the setting to read-only with the key-label "prodDBConnection"-"westUS".</p>
      *
-     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.setReadOnly#string-string}
+     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.setReadOnly#string-string-boolean}
+     *
+     * <p>Clear read-only of the setting with the key-label "prodDBConnection"-"westUS".</p>
+     *
+     * {@codesnippet com.azure.data.appconfiguration.configurationasyncclient.setReadOnly#string-string-boolean-clearReadOnly}
      *
      * @param key The key of configuration setting to set to be read-only.
-     * @param label The label of configuration setting to read-only, or optionally, null if a setting with label is
-     * desired.
+     * @param label The label of configuration setting to read-only. If {@code null} no label will be used.
+     * @param isReadOnly The boolean value to set the setting to read-only or not read-only.
+     * If it is true, set the setting to read-only. If false, the setting won't set to read-only.
```

Comment:
`Flag used to set the read-only status of the configuration. {@code true} will put the configuration into a read-only state, {@code false} will clear the state.`